### DISCUSSION TITLE
API to check if a PR is mergeable

### DISCUSSION THREAD

**Discussion by wkjung** [community/community#25284] [Actions]
<p>Is there way to check (in action) if a PR is mergeable in terms of the required minimum approving reviews and checks that are passed?</p>
<p>Our workflow is like:<br>
submit PR → after some reviewing process two (the required minimum) approving reviews are made and checks are passing → we manually assign a senior who can merge it → she merges it.</p>
<p>We’d like to automate the last two process in github action. As there are many actions that assign, all I need is to know if the two conditions are met and we can merge it. Is there a way to do this?</p>

**Comment by rossjrw** [comment, marked as answer]
<p>The <code>pull_request</code> webhook payload has a <code>mergeable</code> property that may be what you’re looking for:</p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request" class="inline-onebox" rel="noopener nofollow ugc">Webhook events and payloads - GitHub Docs</a></p>
<p>You could create an action that fires on the <code>pull_request_review</code> event:</p>
<p><a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review" class="inline-onebox" rel="noopener nofollow ugc">Events that trigger workflows - GitHub Docs</a></p>
<p><a href="https://docs.github.com/en/actions/learn-github-actions/contexts#github-context" class="inline-onebox" rel="noopener nofollow ugc">Contexts - GitHub Docs</a></p>
<pre data-code-wrap="yml"><code class="lang-nohighlight">on:
  pull_request_review:
    types: ['submitted']
jobs:
  assign-senior-if-mergeable:
    runs-on: ubuntu-latest
    steps:
      - name: Assign senior
        if: github.event.pull_request.mergeable
        run: &lt;some code that assigns the senior&gt;
</code></pre>
<p>Haven’t tested this at all, just written on the fly. You might need a wider set of starting events/event types as just checking for mergeability when a review is submitted might not be sufficient (e.g. to check if the review count is &gt;= 2). You might also need a more intelligent conditional, as ‘mergeable’ might not be precisely what you want (but it might already have the review count check built in).</p>
